### PR TITLE
API wrapper updates

### PR DESCRIPTION
**Pull request by rehan-vanzyl**

### CODE REVIEW

File: lib/mxit_api/client.rb [hunk 1]
```diff
@@ -198,7 +198,11 @@ def recommend_app(callback_url, from_user_id, to_user_id, message, options={ aut
       handle_response(response)
     end
 
-    def batch_notify_users(mxit_ids, message, contains_markup)
+    # When the Mxit API returns 400::Bad Request due to a non-existent Mxit ID in the batch the rest
+    # of the IDs still receive the message.
+    def batch_notify_users(mxit_ids, message, contains_markup, options={ spool: true,
+      spool_timeout: 60*60*24*7 })
```
Comment (linsen) [inline lib/mxit_api/client.rb:204]: I could be wrong, but I think specifying the default parameters here is risky - if someone gives an options hash, but doesn't give these two params they will not be set.  Rather do `options={}`, and then in the body of the method have `options[:spool] ||= true`, etc.

**Comment by linsen** [comment]
Was about to ask about the changelog, but I see it's been updated in a separate request :-P

File: lib/mxit_api/client.rb (hunk 1 above)
Comment (rehan-vanzyl) [inline lib/mxit_api/client.rb:204]: I was unsure about the standard in Ruby? I actually do `spool = options[:spool].nil? ? true : options[:spool]` in `send_message`

Comment (inothnagel) [inline lib/mxit_api/client.rb:204]: This page also lists a few idioms for handling default options: http://blog.leshill.org/blog/2009/06/10/default-options-with-ruby.html

Comment (rehan-vanzyl) [inline lib/mxit_api/client.rb:204]: Cool, will definitely use this in the next iteration.

**Pull request merged**